src/vibrato.rs

```diff
@@ -0,0 +1,457 @@
+            delay_secs,
+            width_secs,
+            mod_freq_hz,
+            num_channels,
+        )
+        .unwrap();
+        let (delay_secs_check, width_secs_check, mod_freq_hz_check) = vibrato.get_param();
+        assert_eq!(delay_secs_check, 0.01);
+        assert_eq!(width_secs_check, 0.005);
+        assert_eq!(mod_freq_hz_check, 5.0);
+    }
+
+    #[test]
+    fn test_set_param() {
+        let sample_rate_hz = 44100.0;
+        let delay_secs = 0.01;
+        let width_secs = 0.005;
+        let mod_freq_hz = 5.0;
+        let num_channels = 2;
+        let mut vibrato = VFilter::new(
+            sample_rate_hz,
+            delay_secs,
+            width_secs,
+            mod_freq_hz,
+            num_channels,
+        )
+        .unwrap();
+        let (delay_secs_check_1, width_secs_check_1, mod_freq_hz_check_1) = vibrato.get_param();
+        vibrato.set_param(0.02, 0.01, 10.0).unwrap();
+        let (delay_secs_check_2, width_secs_check_2, mod_freq_hz_check_2) = vibrato.get_param();
+        assert_eq!(delay_secs_check_1 * 2.0, delay_secs_check_2);
+        assert_eq!(width_secs_check_1 * 2.0, width_secs_check_2);
+        assert_eq!(mod_freq_hz_check_1 * 2.0, mod_freq_hz_check_2);
+    }
+
+    #[test]
+    fn test_reset() {
+        let sample_rate_hz = 44100.0;
+        let delay_secs = 0.01;
+        let width_secs = 0.005;
+        let mod_freq_hz = 5.0;
+        let num_channels = 1;
+        let mut vibrato = VFilter::new(
+            sample_rate_hz,
+            delay_secs,
+            width_secs,
+            mod_freq_hz,
+            num_channels,
+        )
+        .unwrap();
+        let input = vec![vec![1.0; 44100]; 1];
+        let mut output = vec![vec![0.0; 44100]; 1];
+        // Convert Vec<Vec<f32>> to &[&[f32]] for input and &mut [&mut [f32]] for output
+        let input_slices: Vec<&[f32]> = input.iter().map(|v| v.as_slice()).collect();
+        let input: &[&[f32]] = &input_slices;
+
+        let mut output_slices: Vec<&mut [f32]> =
+            output.iter_mut().map(|v| v.as_mut_slice()).collect();
+        let mut output: &mut [&mut [f32]] = &mut output_slices;
+        vibrato.process(&input, &mut output);
+        vibrato.reset();
+        assert_eq!(output[0][0], 0.0);
+    }
+
+    #[test]
+    fn output_equals_delayed_input_with_zero_modulation() {
+        let sample_rate_hz = 3.0;
+        let delay_secs = 1.0; // 10 samples of delay
+        let width_secs = 0.0; // No modulation
+        let mod_freq_hz = 5.0; // Modulation frequency, irrelevant here due to zero width
+        let num_channels = 1;
+        let mut vibrato = VFilter::new(
+            sample_rate_hz,
+            delay_secs,
+            width_secs,
+            mod_freq_hz,
+            num_channels,
+        )
+        .unwrap();
+
+        let input_len = 5;
+        let input = vec![vec![1.0; input_len]; num_channels];
+        let mut output = vec![vec![0.0; input_len]; num_channels];
```

Comment:
The **input_len** need to be larger if your **sample_rate_hz** is 3.0 and the **delay_secs** is 1.0. Might be a good idea to test with a sinusoid input too.